subsys/bluetooth/shell/flash.c

```diff
@@ -0,0 +1,227 @@
+/** @file
+ * @brief Bluetooth Controller and flash co-operation
+ *
+ */
+
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <shell/shell.h>
+#include <misc/printk.h>
+
+#include <stdlib.h>
+#include <string.h>
+#include "flash.h"
+#include <soc.h>
+
+#define FLASH_SHELL_MODULE "flash"
+#define BUF_ARRAY_CNT 16
+#define TEST_ARR_SIZE 0x1000
+
+u8_t test_arr[TEST_ARR_SIZE];
+
+static int cmd_erase(int argc, char *argv[])
+{
+	struct device *flash_dev;
+	u32_t page_addr;
+	int result;
+	u32_t size;
+
+	flash_dev = device_get_binding(CONFIG_SOC_FLASH_NRF5_DEV_NAME);
+
+	if (!flash_dev) {
+		printk("Nordic nRF5 flash driver was not found!\n");
+		return -1;
+	}
+
+	if (argc < 2) {
+		printk("Missing page address.\n");
+		return -1;
+	}
+
+	page_addr = strtoul(argv[1], NULL, 16);
+
+	if (argc > 2) {
+		size = strtoul(argv[2], NULL, 16);
+	} else {
+		size = NRF_FICR->CODEPAGESIZE;
+	}
+
+	flash_write_protection_set(flash_dev, 0);
+
+	result = flash_erase(flash_dev, page_addr, size);
+
+	if (result) {
+		printk("Erase Failed, code %d.\n", result);
+	} else {
+		printk("Erase success.\n");
+	}
+
+	return result;
+}
+
+static int cmd_flash(int argc, char *argv[])
+{
+	u32_t check_array[BUF_ARRAY_CNT];
+	u32_t buf_array[BUF_ARRAY_CNT];
+	struct device *flash_dev;
+	u32_t w_addr;
+	int j = 0;
+
+	flash_dev = device_get_binding(CONFIG_SOC_FLASH_NRF5_DEV_NAME);
+
+	if (!flash_dev) {
+		printk("Nordic nRF5 flash driver was not found!\n");
+		return -1;
+	}
+
+	if (argc < 2) {
+		printk("Missing address.\n");
+		return -1;
+	}
+
+	w_addr = strtoul(argv[1], NULL, 16);
```

Comment:
You could in principle move this assignment further down, after flash_write_protection_set() since that's the first place you use it.